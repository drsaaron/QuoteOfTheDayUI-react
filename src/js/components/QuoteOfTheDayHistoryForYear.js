/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

import React, {Component} from 'react';

export default class QuoteOfTheDayHistoryForYear extends Component {

    render() {
        var year = this.props.year;
        var history = this.props.history;

        return (
                <article>
                    {year}
                    <ul>
                        { history.map(datum => <li key={datum.runDate}>{datum.runDate}</li>) }
                    </ul>
                </article>
                );
    }
}
