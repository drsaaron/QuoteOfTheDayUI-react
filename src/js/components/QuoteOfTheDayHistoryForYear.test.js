import React from 'react';
import renderer from 'react-test-renderer';

import QuoteOfTheDayHistoryForYear from './QuoteOfTheDayHistoryForYear';

describe('quote of the day history for year', () => {
    test('basic', () => {
	const year = 2001;
	const history = [ { runDate: '2001-06-01' }, { runDate: '2001-07-01' } ];
	const component = renderer.create(<QuoteOfTheDayHistoryForYear year={year} history={history} />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
    });
});
