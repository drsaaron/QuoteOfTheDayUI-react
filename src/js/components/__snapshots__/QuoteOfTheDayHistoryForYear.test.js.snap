// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote of the day history for year basic 1`] = `
<article>
  2001
  <ul>
    <li>
      2001-06-01
    </li>
    <li>
      2001-07-01
    </li>
  </ul>
</article>
`;
